### PR TITLE
Add exceptions to List command

### LINKED ISSUE #31: As a careless user, I want to receive an error message when entering a wrong command
... so that I can be aware that I need to rectify my incorrect input

### PR DESCRIPTION
**Pull request by chinhan99**

### CODE REVIEW

File: src/main/java/seedu/duke/command/AddCommand.java [hunk 1]
```diff
@@ -81,6 +82,11 @@ public void execute(TransactionList transactions, Ui ui, Storage storage) throws
         // Throws exception if there are no parameters
         checkParameterExist(splits);
 
+        assert (splits.length == TAG_LIMIT) : "Number of split inputs must equal 5.";
+        // At this stage, the number of split inputs will be == 5 as all the 5 tags must exist
+        // Any additional tags would throw an exception before this assertion.
```
Comment (chydarren) [inline src/main/java/seedu/duke/command/AddCommand.java]: Good assertion over here, note that for single-line comments there will be no period at the end ;)

File: src/main/java/seedu/duke/common/Utilities.java [hunk 2]
```diff
@@ -53,6 +54,22 @@ public static void checkExtraTagExist(String[] splits, int tagLimit) throws Inpu
...
 
+
+    /**
+     * Checks if there are missing parameter within the user input.
+     * If the split.length() is <= 2, it means that only the tag exists , and there is no parameter after the tag.
+     *
+     * @param splits The user input after the command word, split into a list for every space found.
+     * @throws AddTransactionMissingParameterException Extra tag exception.
+     */
+    public static void checkParameterExist(String[] splits) throws AddTransactionMissingParameterException {
```
Comment (chydarren) [inline src/main/java/seedu/duke/common/Utilities.java]: Shall we rename the exception is thrown, probably to InputMissingParameterException();? As the `checkParameterExist()` function it shared by the AddCommand and ListCommand classes.

Comment (chinhan99) [reply src/main/java/seedu/duke/common/Utilities.java]: Good point. Will rename this exception accordingly!

File: src/main/java/seedu/duke/command/FindCommand.java [hunk 3]
```diff
@@ -75,11 +69,13 @@ public void execute(TransactionList transactions, Ui ui, Storage storage) {
         try {
             // Checks the format of find to ensure that it contains keywords used in the search expression
             checkFindFormat(keywords);
+            assert !keywords.isBlank();
```
Comment (brian-vb) [inline src/main/java/seedu/duke/command/FindCommand.java:72]: Here, we are asserting whether keywords is not blank. Should there be an error message should this assertion fails?

**Review by brian-vb — APPROVED** [review]
Other than those assert statements, which shouldn't impact the code, this works!!

File: src/main/java/seedu/duke/command/FindCommand.java (hunk 3 above)
Comment (chydarren) [reply src/main/java/seedu/duke/command/FindCommand.java:72]: Yup an error message will be generated by the assertion automatically ^^

**Review by chydarren — APPROVED** [review]
Looks good, thanks for working on the feedback given

**Pull request merged**